File: 코딩 기초 트레이닝/주사위 게임1-p181839/주사위 게임1.js
// https://school.programmers.co.kr/learn/courses/30/lessons/181839

function solution(a, b) {
    // a and b odd -> a^2 + b^2
    // a or b odd
    // -> 2 * (a + b)
    // a and b even -> | a - b |
    if( a % 2 !== 0 && b % 2 !== 0 ) {
        return a**2 + b **2;
    } else if( a % 2 === 0 && b % 2 ===0) {
        return Math.abs(a-b);
    } else {
        return 2 * (a+b);
    }
}

File: 코딩 기초 트레이닝/주사위 게임1-p181839/주사위 게임1.py
# https://school.programmers.co.kr/learn/courses/30/lessons/181839

def solution(a, b):
    # a and b odd -> a^2 + b^2
    # a or b odd -> 2 * (a + b)
    # a and b even -> | a - b |

    if a % 2 == 0 and b % 2 == 0:
        return abs(a-b)
    elif a % 2 != 0 and b % 2 != 0:
        return (a**2) + (b**2)
    else:
        return 2 * (a+b)
